packages/test/test-end-to-end-tests/src/testSnapshotCache.ts

```diff
@@ -26,9 +26,12 @@ export class TestSnapshotCache implements IPersistedCache {
 	public async put(entry: ICacheEntry, value: ValueWithSnapshot): Promise<void> {
 		const key = getKeyForCacheEntry(entry);
 		this.cache.set(key, value);
-		const versionKey = `${value.value.snapshotTree.id}`;
-		this.versionCache.set(versionKey, value);
-		this.versionToCacheKey.set(versionKey, key);
+
+		if (value.value.snapshotTree !== undefined) {
```

Comment:
What does this do? Can you add a comment.

Reply:
I'm going to move this change to a different PR as there's some really weird funkiness when I use a cache release mechanism. Not sure why. All I know is that if I free up the cache, the test ends early before I can hit the asserts. I tried putting try catch around the `await provider.ensureSynchronized`, but it didn't work.